### DISCUSSION TITLE
Is default value for groupSlug wrong?

### DISCUSSION THREAD

**Discussion by geraintwhite** [renovatebot/renovate#10636] [Request Help]
In the [docs](https://docs.renovatebot.com/configuration-options/#groupslug), the example produces the branch name `renovate/devdependencies-non-major` from the group name "devDependencies (non-major)".

However, when running renovate (latest version under docker) the branch names are as follows:

```
       "branchList": [
         "renovate/copy-webpack-plugin-9.x",
         "renovate/dependencies-(non-major)",
         "renovate/devdependencies-(non-major)",
         "renovate/html-webpack-plugin-5.x",
         "renovate/node-14.x",
         "renovate/npm-7.x",
         "renovate/react-native-modal-12.x",
         "renovate/react-native-reanimated-2.x",
         "renovate/terser-webpack-plugin-5.x",
         "renovate/wait-on-6.x",
         "renovate/webpack-5.x"
       ],
```

With the following config:

```
{
  "extends": [
    "config:base",
    "schedule:weekends",
    ":rebaseStalePrs",
    ":noUnscheduledUpdates",
    ":prNotPending",
    ":dependencyDashboard"
  ],
  "stabilityDays": 3,
  "internalChecksFilter": "strict",
  "enabledManagers": ["npm", "nvm"],
  "node": {
    "supportPolicy": ["lts_latest"],
    "major": { "enabled": true }
  },
  "ignoreDeps": ["flow-bin", "electron"],
  "ignorePaths": ["**/node_modules/**", "**/Pods/**"],
  "packageRules": [
    {
      "groupName": "dependencies (non-major)",
      "matchDepTypes": ["dependencies"],
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": false
    },
    {
      "groupName": "devDependencies (non-major)",
      "matchDepTypes": ["devDependencies"],
      "matchUpdateTypes": ["minor", "patch"],
      "automerge": true
    }
  ]
}

```

**Comment by viceice** [comment]
@rarkins Regression of #10621?

**Reply by geraintwhite** [reply]
I think it's just the default behaviour of https://github.com/simov/slugify to not replace brackets, which contradicts the `groupSlug` docs.

**Reply by geraintwhite** [reply]
Using `slugify(groupName, { lower: true, strict: true })` would solve it.

**Reply by rarkins** [reply]
It seems pretty coincidental with my change, so I'd want to debug it to be sure it wasn't my recent fault. But the change was meant to only apply if you had a rule with a groupSlug matching followed by another matching rule with only groupName. 

Maybe it's like @grit96 suspects and a bug that's existed before. I'd prefer strict mode if that removes brackets.
